### DISCUSSION TITLE
Build Custom Knowledge Base

### DISCUSSION THREAD

**Discussion by KMayank29** [explosion/spaCy#10710] [Help: Other Questions]
Hi Team SpaCy

I have hundreds of articles that generally mention lesser known people and organizations. These persons and organizations don't have wiki pages.
Now, my objective is to build a knowledge base like wikidata/DBpedia and store the entities and their relations in a graph Database.
For example: Peter-->[WORKS_IN]-->Fanboy

Given the datasets and objective I have, could you please guide me or point me to a resource/blog so that I can achieve my objective.

PS: I have started by extracting relationships between Person entity and Org entity by using SpaCy match patterns with fixed set of rules.
Once I have the triplets (PERSON, RELATION, ORG) I populate the Graph DB.
I face issues like person with same names working in same company needs to be disambiguated.

Thanks

**Comment by ljvmiranda921** [comment]
Hi @KMayank29 , 

You can use [this tutorial project](https://spacy.io/universe/project/video-entity-linking) as a guide on building a knowledge base. The only difference here is that in the tutorial, we used Wikidata IDs, which may not be available to you (or you won't be using). 

Our recommendation is for you to build the graph database **first**, populate the nodes and edges with the entities and relationships you need, and then, use the IDs for the entity linker. The ID generation strategy doesn't have to be clever: you can use UUIDs, or reuse the IDs the graph database provides (if there's any).
